language/move-lang/tests/move_check/dependencies/use_friend_transitive_by_use.exp

```diff
@@ -0,0 +1,11 @@
+error: 
+
+    ┌── tests/move_check/dependencies/use_friend_transitive_by_use.move:16:9 ───
+    │
+ 16 │         C::c()
+    │         ^^^^^^ Invalid use of module '0x2::C' in module '0x2::B'.
+    ·
+ 16 │         C::c()
+    │         ------ Using this module creates a dependency cycle: '0x2::C' may use '0x2::A' uses '0x2::B' uses '0x2::C'
```

Comment:
I don't quite like the error message I put here where there is mix of `use` and `friend` edges in the cycle. But I couldn't find a better one-liner message to describe it. @tnowacki @bob-wilson looking for suggestions here.

Reply:
I can't think of anything better offhand, but I would drop the "may" and just say "uses" regardless of whether the use is due to a friend relationship or not.

Reply:
Dropping the "may" sounds fair to me, but it may also hide the actual relation... How do you like,
`'0x2::C' is a friend of '0x2::A' uses '0x2::B' uses '0x2::C'`? (i.e., "may use" -> "is a friend of")

Reply:
Does the distinction matter? I would expect that someone seeing this message would open the file and search for the module name, and then they can see how it is used.

I don't really mind "is a friend of" but honestly I don't particularly like the original message. It should have "which uses" instead of just "uses". But.... that gets pretty verbose. Perhaps an alternative would be:

"The dependency cycle is: '0x2::C', '0x2::A', '0x2::B', '0x2::C'"

Maybe that's too terse but it would avoid the question of "uses" vs. "is a friend of"

Reply:
> Does the distinction matter? I would expect that someone seeing this message would open the file and search for the module name, and then they can see how it is used.

I would think the same, and the distinction only matters a little bit on which file to look at:

- if `A uses B`, then the user should open A (the first file) and realize that some function in B is called or some type in B is used.
- if `A may use B`, then the user should actually open B (the second file) and realize that A is declared in the friend list.